Android extra flags not applied on initial CMake generation

cmannett85-arm opened the pull request:
`CMAKE_CXX_FLAGS` was overwritten if not already in the cache (i.e. when first built).  Presumably down to toolchain loading order.

baldurk reviewed (changes requested):
Can you please keep the definition of `ANDROID`? It's possible some past or future NDK won't define that, and a redundant definition does no harm for those that do.

cmannett85-arm committed:
Android extra flags not applied on initial CMake generation

CMAKE_CXX_FLAGS was overwritten if not already in the cache (i.e. when first built).  Presumably down to toolchain loading order.
